ui/app/src/components/DashboardList/RecentDashboardList.tsx

```diff
@@ -0,0 +1,191 @@
+// Copyright 2023 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { DashboardResource } from '@perses-dev/core';
+import { Box, Stack } from '@mui/material';
+import { GridColDef, GridActionsCellItem, GridRowParams, GridValueGetterParams } from '@mui/x-data-grid';
+import DeleteIcon from 'mdi-material-ui/DeleteOutline';
+import PencilIcon from 'mdi-material-ui/Pencil';
+import { useCallback, useEffect, useMemo, useState } from 'react';
+import { dashboardDisplayName } from '@perses-dev/core/dist/utils/text';
+import { DeleteDashboardDialog } from '../DeleteDashboardDialog/DeleteDashboardDialog';
+import { RenameDashboardDialog } from '../RenameDashboardDialog/RenameDashboardDialog';
+import { DashboardDataGrid, Row } from './DashboardDataGrid';
+
+export interface DatedDashboards {
+  dashboard: DashboardResource;
+  date: string;
+}
+
+export interface RecentDashboardListProperties {
+  dashboardList: DatedDashboards[];
+  hideToolbar?: boolean;
+}
+
+export function RecentDashboardList(props: RecentDashboardListProperties) {
+  const { dashboardList, hideToolbar } = props;
+
+  const getDashboard = useCallback(
+    (project: string, name: string) => {
+      return dashboardList.find(
+        (datedDashboard) =>
+          datedDashboard.dashboard.metadata.project === project && datedDashboard.dashboard.metadata.name === name
+      )?.dashboard;
+    },
+    [dashboardList]
+  );
+
+  const [rows, setRows] = useState<Row[]>([]);
+  useEffect(() => {
+    setRows(
+      dashboardList.map(
+        (datedDashboard) =>
+          ({
+            project: datedDashboard.dashboard.metadata.project,
+            name: datedDashboard.dashboard.metadata.name,
+            displayName: dashboardDisplayName(datedDashboard.dashboard),
+            version: datedDashboard.dashboard.metadata.version,
+            createdAt: datedDashboard.dashboard.metadata.created_at,
+            updatedAt: datedDashboard.dashboard.metadata.updated_at,
+            viewedAt: datedDashboard.date,
```

Comment:
I didn’t realize we stored this recently viewed dashboards data, is that what’s happening / can I get a quick summary of how this works? 

Screenshot looks super cool btw 🚀

Reply:
Ohh I see now it’s coming from Local Storage! Again, planning to pull this locally tomorrow so I’ll give better feedback soon

Reply:
Yes, it's coming from local storage 😄 Each time you go to a dashboard view, it will be added in a queue with dashboard project, dashboard name and the current date. The queue has a limit to 100 dashboards (it can be changed, if we want more or less, browsers have size limit for local storage). When reaching 100 dashboards, oldest dashboards are removed. The queue is then saved in the local storage.

Reply:
perhaps 10 dashboards for the project view will be enough :)